code/clue.js

```diff
@@ -41,19 +189,73 @@ const randomSelector = array => {
+  killer:'',
+  weapon:'',
+  room:''
+}
+
+
+
+
 // This function will be invoked when you click on the killer card.
+
 const pickKiller = () => {
   // This will randomly select a killer from the suspects. And add that to the mystery object.
   mystery.killer = randomSelector(suspects)
+  
+  // Shuffle weapon function
+  shuffleFavouriteWeapon = () => {
+    mystery.killer.favouriteWeapon = randomSelector(weapons); 
+  };
+    shuffleFavouriteWeapon();
+    console.log(mystery.killer.favouriteWeapon.name)
 
   // This will change the background color of the card to the one connected to the chosen killer and show the full name of the killer. Feel free to add more things to show about the killer.
   document.getElementById('killerCard').style.background = mystery.killer.color
+  document.getElementById('killerImage').src = mystery.killer.image
   document.getElementById(
     'killerName'
   ).innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`
+  document.getElementById('killerFavoriteWeapon').innerHTML = `${mystery.killer.favouriteWeapon.name}`
 }
+document.getElementById('killerCard').addEventListener('click', pickKiller)
+
+//  pickWeapon 
+const pickWeapon = () => {
+  // This will randomly select a killer from the suspects. And add that to the mystery object.
+  mystery.weapon = randomSelector(weapons)
+
+  // This will change the background color of the card to the one connected to the chosen killer and show the full name of the killer. Feel free to add more things to show about the killer.
+  document.getElementById('weaponCard').style.background = mystery.killer.color
+  document.getElementById(
+    'weaponName'
+  ).innerHTML = `${mystery.weapon.name}`
+  document.getElementById(
+    'weaponWeight'
+  ).innerHTML = `${mystery.weapon.weight}`
+  
+}
+document.getElementById('weaponCard').addEventListener('click', pickWeapon)
+
+// pickRoom
+const pickRoom = () => {
+  // This will randomly select a killer from the suspects. And add that to the mystery object.
+  mystery.room = randomSelector(rooms)
+
+  // This will change the background color of the card to the one connected to the chosen killer and show the full name of the killer. Feel free to add more things to show about the killer.
+  document.getElementById('roomCard').style.background = mystery.killer.color
+  document.getElementById(
+    'roomName'
+  ).innerHTML = `${mystery.room}`
+}
+document.getElementById('roomCard').addEventListener('click', pickRoom)
 
-// CREATE FUNCTIONS pickWeapon and pickRoom in a similar way.
 
 // STEP 4 - CREATE A FUNCTION revealMystery that will be invoked when you click that button. It should show something like:
 // 'The murder was committed by Jacob Green, in the living room with a rope.'
+
+
+const revealMystery = () => {
+  document.getElementById('mystery').innerHTML = `The murder was committed by ${mystery.killer.firstName} ${mystery.killer.lastName}, in the ${mystery.room} with a ${mystery.weapon.name}`
+}
```

Comment:
You forgot to block suspect, weapon and room cards functionality over here - even after revealing crime I was able to change suspect card